app/src/pages/api/report/[id].ts

```diff
@@ -0,0 +1,92 @@
+import {
+    BAD_REQUEST,
+    HttpMethod,
+    IApiErrorResponse,
+    IReport,
+    INTERNAL_SERVER_ERROR,
+    METHOD_NOT_ALLOWED,
+    NOT_FOUND,
+} from '@/models';
+import { ReportDbClient, isUndefined } from '@/utils';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+export interface IGetReportResponse {
+    report: IReport;
+}
+
+/**
+ * Get or update existing report
+ *
+ * Allowed methods: GET, PUT
+ *
+ * Parameters:
+ *  - id: string, report ID
+ *  - body: IReport, new report to pUT
+ *
+ * Response: IGetReportResponse
+ *
+ * Potential errors:
+ *  - 400: when body is invalid or not provided
+ *  - 404: when report does not exist
+ *  - 405: when non-allowed method is used
+ *  - 500: internal server error
+ */
+export default async function handler(
+    req: NextApiRequest,
+    res: NextApiResponse<IGetReportResponse | IApiErrorResponse>
+) {
+    if (req.method !== HttpMethod.GET && req.method !== HttpMethod.PUT) {
+        return res.status(405).send({ report: METHOD_NOT_ALLOWED });
+    }
+
+    const { id } = req.query;
+
+    try {
+        const reportClient = new ReportDbClient();
+
+        if (req.method === HttpMethod.PUT) {
+            // PUT request: store the payload in the database
+            const params = req.body as IReport;
+            const isValidRequest =
```

Comment:
Are these all really mandatory? I'm okay merging this initially, but we should have more nuanced validations